airbyte-integrations/connectors/source-pinterest/source_pinterest/source.py

```diff
@@ -88,27 +89,38 @@ def streams(self, config: Mapping[str, Any]) -> List[Stream]:
         report_config = self._validate_and_transform(config, amount_of_days_allowed_for_lookup=913)
 
         declarative_streams = super().streams(config)
-        ad_accounts = [stream for stream in declarative_streams if stream.name == "ad_accounts"][0]
 
         # Report streams involve async data fetch, which is currently not supported in low-code
         report_streams = [
-            CampaignAnalyticsReport(ad_accounts, config=report_config),
-            CampaignTargetingReport(ad_accounts, config=report_config),
-            AdvertiserReport(ad_accounts, config=report_config),
-            AdvertiserTargetingReport(ad_accounts, config=report_config),
-            AdGroupReport(ad_accounts, config=report_config),
-            AdGroupTargetingReport(ad_accounts, config=report_config),
-            PinPromotionReport(ad_accounts, config=report_config),
-            PinPromotionTargetingReport(ad_accounts, config=report_config),
-            ProductGroupReport(ad_accounts, config=report_config),
-            ProductGroupTargetingReport(ad_accounts, config=report_config),
-            KeywordReport(ad_accounts, config=report_config),
-            ProductItemReport(ad_accounts, config=report_config),
-        ] + self.get_custom_report_streams(ad_accounts, config=report_config)
+            CampaignAnalyticsReport(self._create_ad_accounts_stream(config), config=report_config),
```

Comment:
@maxi297 and I discussed this aspect a little bit. Although its not great to have to repeat this action so many times to generate independent streams, it doesn't have an I/O on the network and we only do this at the beginning so the performance impact is minimal.

We did play around with deep copy which didn't quite work on the nested objects among other reasons.

We floated a few ideas on how we might address this for other sources that do have dependencies on the API
- a light copy of a stream that doesn't share state
- a full custom copy of a stream
- a way to extract only a single stream from the manifest or something

either way, this is not worth blocking on, but something we'll need to at some point solve for